tests/e2e/framework/utils.go

```diff
@@ -384,6 +386,139 @@ func (f *Framework) AssertMetricsEndpointUsesHTTPVersion(endpoint, version strin
+
+	cmd = exec.Command(ocPath,
+		"adm", "policy", "add-cluster-role-to-user", "cluster-monitoring-view", "-z", "prometheus-query-sa", "-n", f.OperatorNamespace)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to add cluster role to user: %v", err)
+	}
+	return nil
+}
+
+// CleanupRBACForMetricsTest deletes the service account
+func (f *Framework) CleanUpRBACForMetricsTest() error {
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return fmt.Errorf("Failed to find oc binary: %v", err)
+	}
+	cmd := exec.Command(ocPath,
+		"delete", "sa", "prometheus-query-sa", "-n", f.OperatorNamespace)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to delete service account: %v", err)
+	}
+	return nil
+}
+func getCurlPrometheusCMD() string {
+	return `TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token) && { curl -k -s https://prometheus-k8s.openshift-monitoring.svc.cluster.local:9091/api/v1/targets --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $TOKEN"; }`
+}
+
+func (f *Framework) GetPrometheusMetricTargets() ([]promv1.Target, error) {
+	var metricsTargets []promv1.Target
+
+	namespace := f.OperatorNamespace
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return metricsTargets, fmt.Errorf("error getting oc path %s", err)
+	}
+	cmd := exec.Command(ocPath,
+		"run", "--rm", "-i", "--restart=Never", "--image=registry.fedoraproject.org/fedora:latest",
+		"-n", namespace, "--overrides={\"spec\": {\"serviceAccountName\": \"prometheus-query-sa\"}}", "metrics-test", "--", "bash", "-c",
+		getCurlPrometheusCMD(),
+	)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return metricsTargets, fmt.Errorf("error getting output %s", err)
+	}
+
+	outTrimmed := trimOutput(string(out))
+	if outTrimmed == "" {
+		return metricsTargets, fmt.Errorf("error getting output")
+	}
+
+	log.Printf("metrics output:\n%s\n", out[0:1000])
+
+	var responseData struct {
+		Data struct {
+			ActiveTargets []promv1.Target `json:"activeTargets"`
+		} `json:"data"`
+	}
+	err = json.Unmarshal([]byte(outTrimmed), &responseData)
+	if err != nil {
+		return metricsTargets, fmt.Errorf("error unmarshalling json %s", err)
+	}
+
+	// Filter metrics for the specified namespace
+	for _, metricsTarget := range responseData.Data.ActiveTargets {
+		// check if there is a metric label first
+		if metricsTarget.Labels != nil {
+			if checkMetricTargetLabelValue(metricsTarget, "namespace", namespace) {
+				// check if it has endpoint equal to metrics or metrics-fio
+				if checkMetricTargetLabelValue(metricsTarget, "endpoint", "metrics") || checkMetricTargetLabelValue(metricsTarget, "endpoint", "metrics-co") {
+					metricsTargets = append(metricsTargets, metricsTarget)
+				}
+			}
+
+		}
+
+	}
+
+	return metricsTargets, nil
+}
+
+// function to check a label value in a metric match certain value
+func checkMetricTargetLabelValue(metricTarget promv1.Target, labelName string, labelValue string) bool {
```

Comment:
nit: Since this returns a boolean, we have the opportunity to name it for readability if we do something like `metricContainsLabel`.

Making the calling code:

```go
  if metricContainsLabel(metric, name, value) {
    ..
  }
```